### ISSUE TITLE
Experiment: promoters YOLO run

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/bolinas-dna#21]
## Description
Train a model of a modest size (~1B) with reasonable defaults on an animal promoters dataset.

## Hypothesis or Goal
See how far you can get in downstream tasks and understand major strengths and weaknesses to guide future experiments.

### Links
[Training code](https://github.com/marin-community/marin/blob/b5cb8b6139d0d0e32f9499a97723dc19a2d27617/experiments/dna/exp21_promoters_yolo.py)
[Analysis code](https://github.com/Open-Athena/bolinas-dna/tree/main/snakemake/analysis/evals_v1)
[Wandb](https://wandb.ai/gonzalobenegas/marin/runs/animal-promoters-yolo-r01-213a8e?nw=nwusergonzalobenegas)

### Variant effect prediction
Setup:
- Classification of causal variants for human Mendelian and complex traits (TraitGym).
  - (I'm also working on larger versions of these datasets, as part of the TraitGym paper revision).
- Only odd chromosomes (validation set).
- Two types of variants roughly overlapping promoter regions the model was trained on: promoters and 5' UTRs.

Results:
- Typically matches the performance of Evo 2 (40B).
  - Not in complex/promoter. Understanding these variants perhaps benefits from understanding other regions of the genome.
- Still generally behind GPN-Star, most clearly in complex traits.
<img width="1127" height="1145" alt="Image" src="https://github.com/user-attachments/assets/0029c6de-3a1f-4d47-93e8-94c882e4dca8" />

- Performance seems to plateau while pre-training loss continues to go down.
<img width="300" height="300" alt="Image" src="https://github.com/user-attachments/assets/5048d53f-9276-4f26-8515-d42dd005a9ba" />

## Follow up ideas
- Intuitively we should be able to obtain better performance by specializing the model (trained on all animals) into a closer evolutionary neighborhood to human (e.g. primates or mammals). We know this is particularly important for complex trait variants. It would be interesting to try different approaches, such as finetuning just on primates vs. on all animals but spike up the proportion of primates.

- Beyond zero-shot tasks, could we find/develop a good transfer learning benchmark for promoter models? The tasks in [PromoterAI paper](https://www.science.org/doi/10.1126/science.ads7373) look interesting.

**Comment by gonzalobenegas** [comment]
🤖 This model was later evaluated on TraitGym Mendelian **v2** (validation set, odd chromosomes — same v2 eval used in #55, #58, #59), but only for a single checkpoint: **step 16000**.

**AUPRC at step 16000 (`minus_llr`)** — from `s3://oa-bolinas/snakemake/analysis/evals_v1/results/metrics/traitgym_mendelian_v2/animal_promoters_yolo_repeat_downweight/16000.parquet`:

| subset | AUPRC | n_pos | n_neg |
|---|---:|---:|---:|
| tss_proximal | **0.579** | 180 | 1620 |
| 5_prime_UTR_variant | 0.529 | 210 | 1890 |
| splicing | 0.303 | 267 | 2403 |
| synonymous_variant | 0.244 | 46 | 414 |
| non_coding_transcript_exon_variant | 0.238 | 115 | 1035 |
| global | 0.233 | 2453 | 22077 |
| missense_variant | 0.148 | 1480 | 13320 |
| 3_prime_UTR_variant | 0.124 | 77 | 693 |
| distal | 0.120 | 78 | 702 |

**Comparison with the #55 evolutionary-timescale sweep (all 0.6B, 256 bp context):** this model (1.7B, **512 bp** context, trained on animal promoters) is the best *animal*-scope model on promoters by a wide margin — on `tss_proximal` it reaches **0.579 vs 0.342** for [`exp55-animals-r01-c5dd2f`](https://github.com/Open-Athena/bolinas-dna/issues/55). It's also the best overall on `tss_proximal` across every model evaluated on v2 (edges out `exp55-mammals-r01-52f413` at 0.568). On `5_prime_UTR_variant` the gap closes — it's competitive (0.529) but below `exp55-vertebrates-r01-64b0da` (0.576) and `exp55-mammals-r01-52f413` (0.549).

Caveat: the model size (1.7B vs 0.6B) and context length (512 vs 256 bp) both differ from the #55 sweep, so it's not a clean evolutionary-timescale comparison — just a landmark for "what's achievable on v2 with a larger animal-trained promoter model".

**Comment by gonzalobenegas** [comment]
🤖 Updated plot using the post-#194 evals_v2 dataset (`mendelian_traits`, hf_revision `4aed58e`).

![exp21_promoters_yolo_auprc](https://gist.githubusercontent.com/gonzalobenegas/3649e68fb63ca1f3443e4486078eb4d8/raw/c785daa5a1a938365ace6861791fa3e978ac53cf/exp21_promoters_yolo_auprc.svg)

Recipe: [`plots/exp21_promoters_yolo.py`](https://github.com/Open-Athena/marin-dna/blob/0058ec4779401307a9b3a3c9218127511711bce4/plots/exp21_promoters_yolo.py)
